WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java

```diff
@@ -376,7 +386,40 @@ public void onRefreshRequested(StatsService.StatsEndpointsEnum[] endPointsNeedUp
         if (mIsUpdatingStats) {
             return;
         }
-        refreshStats(mCurrentTimeframe, mRequestedDate, true);
+
+        // fragments that needs an update require it at almost the same time. Keep a list of fragments
+        // that require the update and call the service with the right parameters.
+        synchronized (fragmentsRefreshListSynchObj) {
+           for (int i = 0; i < endPointsNeedUpdate.length; i ++) {
+               StatsService.StatsEndpointsEnum current = endPointsNeedUpdate[i];
+               if (!fragmentsRefreshList.contains(current)) {
+                   fragmentsRefreshList.add(current);
+               }
+           }
+
+           // the last fragment set the right call
+           if (fragmentsRefreshListTimer != null)  {
+               fragmentsRefreshListTimer.cancel();
+           }
+
+           fragmentsRefreshListTimer = new Timer();
+           fragmentsRefreshListTimer.schedule(new TimerTask() {
```

Comment:
A TimerTask seems an odd choice for a one-shot deal like this when a handler with postDelayed() can do the same thing [without the drawbacks](https://www.google.com/?q=android+timer+vs+handler#q=android+timer+vs+handler) of a TimerTask.